src/rust.rs

```diff
@@ -172,11 +173,31 @@ pub enum JSEngineError {
 /// A handle that must be kept alive in order to create new Runtimes.
 /// When this handle is dropped, the engine is shut down and cannot
 /// be reinitialized.
-pub struct JSEngine(());
+pub struct JSEngine {
+    /// The count of alive handles derived from this initialized instance.
+    outstanding_handles: Arc<AtomicU32>,
+    // Ensure this type cannot be sent between threads.
```

Comment:
Could `impl !Send for JSEngine{}` be used for that purpose?

Reply:
```
error[E0658]: negative trait bounds are not yet fully implemented; use marker types for now
   --> src/rust.rs:181:1
    |
181 | impl !Send for JSEngine {}
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: for more information, see https://github.com/rust-lang/rust/issues/13231
    = help: add `#![feature(optin_builtin_traits)]` to the crate attributes to enable
```
I'd rather not introduce nightly-only features to this crate.